src/ir/fileReader.cpp

```diff
@@ -245,6 +300,7 @@ bool loadFromFile(Context* c, string filename,Module** top) {
 
     //If top exists return it
     if (top && j.count("top")) {
+      c->getNamespace("global")->print();
```

Comment:
`loadFromFile` is now over 200 lines long, you should consider decomposing it into smaller functions to keep things better organized. It's hard for me to understand these changes because they touch so many different parts of the process. If each chunk was contained in a function with a well scoped and clearly defined purpose, it would be easier to understand what each change was responsible for.